taca/illumina/HiSeqX_Runs.py

```diff
@@ -71,41 +74,57 @@ def demultiplex_run(self):
                 return False
             logger.info(("Created SampleSheet.csv for Flowcell {} in {} ".format(self.id, samplesheet_dest)))
         ##SampleSheet.csv generated
+
+
         ##when demultiplexing SampleSheet.csv is the one I need to use
+        ## Need to rewrite so that SampleSheet_0.csv is always used.
         self.runParserObj.samplesheet  = SampleSheetParser(os.path.join(self.run_dir, "SampleSheet.csv"))
-
+        #we have 10x lane - need to split the  samples sheet and build a 10x command for bcl2fastq
+        Complex_run = False
+        if len(lanes_10X) and len(lanes_not_10X):
+            Complex_run = True
+
+        if Complex_run:
+            samplesheet_dest_not_10X="SampleSheet_0.csv"
+            with open(samplesheet_dest_not_10X, 'wb') as fcd:
+                fcd.write(_generate_samplesheet_subset(self.runParserObj.samplesheet, lanes_not_10X))
+            samplesheet_dest_10X="SampleSheet_1.csv"
+            with open(samplesheet_dest_10X, 'wb') as fcd:
+                fcd.write(_generate_samplesheet_subset(self.runParserObj.samplesheet, lanes_10X))
+        else:
+            with chdir(self.run_dir):
+                shutil.copy("SampleSheet.csv", "SampleSheet_0.csv")
+        
         per_lane_base_masks = self._generate_per_lane_base_mask()
         max_different_base_masks =  max([len(per_lane_base_masks[base_masks]) for base_masks in per_lane_base_masks])
         if max_different_base_masks > 1:
             # in a HiSeqX run I cannot have different index sizes in the SAME lane
             logger.error("In FC {} found one or more lane with more than one base mask (i.e., different index sizes in \
                          in the same lane".format(self.id))
             return False
-        #I have everything to run demultiplexing now.
-        logger.info('Building bcl2fastq command')
 
+        bcl2fastq_cmd_counter = 0
```

Comment:
Maybe this part can be made more similar to https://github.com/vezzi/TACA/blob/master/taca/illumina/HiSeq_Runs.py#L361 look at the code between 361 and 284, whee with a counter we can demux without prior knowledge of how many commands we need to run. Right now we assume we have at most 2 commands ... it would be great if we can find a similar approach to handle the HiSeq and the HiSeqX case